### PR TITLE
Plugins Catalog: add signature information to details

### LINKED ISSUE #37930: Add signature information on the plugin details page.

### PR DESCRIPTION
**Pull request by leventebalogh**
**Which issue(s) this PR fixes**: https://github.com/grafana/grafana/issues/37930

**What this PR does / why we need it**: 
Adds signature information to the plugins details page under the plugins catalog, and also to the Enterprise plugin list items.

https://user-images.githubusercontent.com/9974811/130811185-beee4681-149e-484e-a674-f417e8eb1267.mov

 

Fixes #37930

---

### ⚠️ &nbsp; Notes
Tests are coming.

### CODE REVIEW

File: public/app/features/plugins/admin/components/PluginListBadges.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ function EnterpriseBadge({ id }: { id: string }) {
 
   return (
     <HorizontalGroup>
+      <PluginSignatureBadge status={plugin.signature} />
```
Comment (jackw) [inline public/app/features/plugins/admin/components/PluginListBadges.tsx:41]: Is it possible to have unsigned Enterprise plugins? I'd decided previously not to include the `PluginSignatureBadge` here as I'd assumed the lock icon on the Enterprise badge was acting like a "signature" type.

Comment (leventebalogh) [reply public/app/features/plugins/admin/components/PluginListBadges.tsx:41]: I see mate. Yea, I think unfortunately it is possible.
IIRC the backend part of a enterprise plugin won't launch if the plugin is not signed, but the frontend part will.

Although maybe they have changed this in Grafana Enterprise, will double check 👍

**Review by mckn — COMMENTED** [review]
I think this looks great! Awesome job @leventebalogh. I added a couple of nits but this works as a charm 🙌🏻

**Review by jackw — APPROVED** [review]
Awesome work on this @leventebalogh. 🚀 

Left a nit about where the enum lives.

I think we could do a follow up PR where we look at making better use of the CatalogPlugin, CatalogPluginDetails and versionSignature info from the gcom endpoint so we can display the signature badges for marketplace plugins within the PluginDetails page.

File: public/app/features/plugins/admin/constants.ts [hunk 2]
```diff
@@ -1,2 +1,8 @@
 export const API_ROOT = '/api/plugins';
 export const GRAFANA_API_ROOT = '/api/gnet';
+export enum PLUGIN_TAB_LABELS {
```
[the remark this answers is not in the document]
Comment (leventebalogh) [reply public/app/features/plugins/admin/constants.ts:3]: You are right. As we discussed I will move this to the `types.ts` file for now, and then we can figure out later if we would like to have all the enums in the `constants.ts` file or not. 👍

**Pull request merged**